checkov/common/bridgecrew/platform_integration.py

```diff
@@ -382,45 +378,58 @@ def _persist_file(self, full_file_path, s3_file_key):
-        if not self.bc_id_mapping:
-            self.get_checkov_mapping_metadata()
-        return self.bc_id_mapping
-
-    def get_ckv_to_bc_id_mapping(self) -> Dict[str, str]:
-        if not self.ckv_to_bc_id_mapping:
-            self.get_checkov_mapping_metadata()
-        return self.ckv_to_bc_id_mapping
-
-    def get_checkov_mapping_metadata(self) -> None:
-        if self.bc_skip_mapping is True:
-            logging.debug("Skipped mapping API call")
-            self.ckv_to_bc_id_mapping = {}
+    def get_platform_run_config(self):
+        if self.skip_download is True:
+            logging.debug("Skipping downloading configs from platform")
             return
-        guidelines_url = self.guidelines_api_url
-        headers = get_user_agent_header()
+
+        if self.is_integration_configured():
+            self.get_customer_run_config()
+        else:
+            self.get_public_run_config()
+
+    def get_customer_run_config(self) -> None:
+        if self.skip_download is True:
+            logging.debug("Skipping getting run config")
+            return
+
+        if not self.bc_api_key or not self.is_integration_configured():
+            raise Exception("Tried to get customer run config, but the API key was missing or the integration was not set up")
+
         try:
-            if (self.bc_api_key is not None):
-                guidelines_url = self.customer_all_guidelines_api_url
-                headers = merge_dicts(headers, {"Authorization": self.get_auth_token(), "Content-Type": "application/json"})
+            token = self.get_auth_token()
+            headers = merge_dicts(get_auth_header(token), get_default_get_headers(self.bc_source, self.bc_source_version))
             if not self.http:
                 self.setup_http_manager()
-            request = self.http.request("GET", guidelines_url, headers=headers)
-            response = json.loads(request.data.decode("utf8"))
-            self.guidelines = response["guidelines"]
-            self.bc_id_mapping = response.get("idMapping")
-            self.ckv_to_bc_id_mapping = {ckv_id: bc_id for (bc_id, ckv_id) in self.bc_id_mapping.items()}
-            logging.debug("Got checkov mappings from Bridgecrew BE")
+            request = self.http.request("GET", self.platform_run_config_url, headers=headers)
+
+            self.customer_run_config_response = json.loads(request.data.decode("utf8"))
+
+            # self.guidelines = response["guidelines"]
+            # self.bc_id_mapping = response.get("idMapping")
+            # self.ckv_to_bc_id_mapping = {ckv_id: bc_id for (bc_id, ckv_id) in self.bc_id_mapping.items()}
+            logging.debug("Got customer run config from Bridgecrew BE")
         except Exception as e:
-            logging.debug(f"Failed to get the guidelines from {guidelines_url}, error:\n{e}")
-            self.ckv_to_bc_id_mapping = {}
+            logging.debug(f"Failed to get the guidelines from {self.guidelines_api_url}, error:\n{e}")
```

Comment:
@mikeurbanski1 This means we have an issue in prod, would upgrade the level here to info / warning